### PR TITLE
luci-mod-admin-full: add SNR tweak option for DSL

### PR DESCRIPTION
**Pull request by andreamerello**
Since 860e053b29a77897d635c1e33007552561062961 I've added to LEDE the ability of tweaking  downstream SNR margin on Lantiq DSL chips via UCI.

This patch adds this option to LUCI. AFAIK Lantiq modems are the only supported by LEDE, so I'm adding the option to the UI unconditionally.

Note: currently the LEDE patch affects only ADSL (not VDSL - regardless of the chipset type), but it will probably be extended; so I didn't care about disabling the option in UI when the modem is in VDSL mode.

### CODE REVIEW

**Comment by jow-** [comment]
Implemented with changes in f0716ecd6e807799c8e16bea72d6ea98ec803d1b - thanks!

**Comment by Uka-Uka** [comment]
@andreamerello Does this still affect ADSL only?

**Comment by andreamerello** [comment]
TL;DR: I don't know, sorry.

When I did this - years ago - I only had an ADSL line, and I could only test only on that.. I don't remember why I had feeling that this won't work on VDSL.. Can't say if I was reasonably sure, or if it was just because it was untested..

BTW (Un)fortunately I've switched to optical fiber connection, and I no longer use that device, nor I have any compatible line to test with...

Andrea

**Comment by Uka-Uka** [comment]
Thanks for your swift reply! 

Actually, I've seen some people on the forum reporting that they managed to stabilize their VDSL line through this option, so maybe it's working after all.

I don't know if you can shed some light on how this work in the first place, I mean I see people setting it to a negative value, why would one do that? isn't true that the higher SNR margin the better line stability?

**Comment by andreamerello** [comment]
AFAIK, the higher the SNR margin, the more stable the line, but also the slower is the speed. In my knowledge the SNR should be always positive in reald world; i'd expect values like > 6..

If they achieve better stability by increasing the target SNR at the cost of lower speed, and vice-versa, then it's probably working. 

Note that, while evaluating the stability of a line might be not so straightforward (i.e. one might experience few disconnections per day), the speed of the line should be numerically reported by the DSL modem itself, so I suggest to look at it while tweaking SNR margin. Also, I don't remember if the DSL modem reports the actual measured SNR; if this is the case then it is easy to check if it's follows the chosen target..

Said that, I don't know what might happen setting the target SNR to a negative value; maybe the DSL SW stack interprets it as a special meaning (i.e. automatic), but this is just a random speculation..
